examples/customize-extractor-error/src/main.rs

```diff
@@ -5,15 +5,13 @@
 //! ```
 
 use axum::{
-    async_trait,
```

Comment:
I don't think we should change this example. By doing that we are sorta saying "this is the best/recommended way to solve this" which I'm not sure is right.

However I do think we should link to `WithRejection` so people know that it's an alternative.

Reply:
Maybe we should keep both of them? `customize-extractor-error` for `WithRejection` and `custom-extractor` for the old example? `WithRejection` is perfect for just replacing the rejection message, but the old example was better if you wanted to access the request + async

Reply:
Imo linking to it is fine since the docs contains an example.

Reply:
And to be clear I'm not saying we necessarily should rename it. I'm was just wondering if `MapRejection` is clearer.

Reply:
`MapRejection` was my first idea but I then proposed `WithRejection` because `.map_` (and correponding `Map` types) to me imply some sort of closure-accepting API.

Reply:
Oh sorry I commented in the wrong place :P

Reply:
> What would they do?

Tbh `FilterRejection` doesn't sound interesting to me, but some kind of `EitherRejection` does (dispatching different variants to different rejections, for example). Nonetheless, I'll do some prototyping and see if any of these are actually useful